tests/test_constellation.py

```diff
@@ -0,0 +1,75 @@
+import io
+from contextlib import redirect_stdout
+
+import docker
+from constellation import docker_util
+
+from wodin_deploy.config import WodinConfig
+from wodin_deploy.wodin_constellation import WodinConstellation
+
+
+def get_site_container_name(site, cfg):
+    return f"{cfg.container_prefix}-{cfg.wodin['name']}-{site}"
+
+
+def pull_all_necessary_images(cl):
+    cl.images.pull("library/redis:6.0")
+    cl.images.pull("mrcide/odin.api:main")
+    cl.images.pull("mrcide/wodin-proxy:latest")
+    cl.images.pull("mrcide/wodin:main")
```

Comment:
constellation does not pull the necessary images automatically if they do not exist so just pulling them to make sure

Reply:
Doesn't it? Do we need to deal with that in the main deploy code rather than just in the test? Or will that be part of the cli?

Reply:
we dont need to deal with it in the main deploy code, this was i think the case with constellation too, constellation does not automatically docker pull but docker on cli does pull automatically. I think for the server this is alright, this is more of a fix for github actions i believe